azkaban-common/src/main/java/azkaban/storage/HdfsStorage.java

```diff
@@ -17,28 +17,78 @@
 
 package azkaban.storage;
 
+import azkaban.AzkabanCommonModuleConfig;
 import azkaban.spi.Storage;
+import azkaban.spi.StorageException;
 import azkaban.spi.StorageMetadata;
+import com.google.common.io.Files;
 import com.google.inject.Inject;
 import java.io.File;
+import java.io.IOException;
 import java.io.InputStream;
+import java.net.URI;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+
+import static com.google.common.base.Preconditions.*;
+import static java.util.Objects.*;
 
 
 public class HdfsStorage implements Storage {
+  private static final Logger log = Logger.getLogger(HdfsStorage.class);
+  private static final String HDFS_SCHEME = "hdfs";
+
+  private final HdfsAuth hdfsAuth;
+  private final URI rootUri;
+  private final FileSystem hdfs;
 
   @Inject
-  public HdfsStorage() {
+  public HdfsStorage(HdfsAuth hdfsAuth, FileSystem hdfs, AzkabanCommonModuleConfig config) {
+    this.hdfsAuth = requireNonNull(hdfsAuth);
+    this.hdfs = requireNonNull(hdfs);
 
+    this.rootUri = config.getHdfsRootUri();
+    requireNonNull(rootUri.getAuthority(), "URI must have host:port mentioned.");
+    checkArgument(HDFS_SCHEME.equals(rootUri.getScheme()));
   }
 
   @Override
-  public InputStream get(String key) {
-    throw new UnsupportedOperationException("Method not implemented");
+  public InputStream get(String key) throws IOException {
+    hdfsAuth.authorize();
+    return hdfs.open(new Path(rootUri.toString(), key));
   }
 
   @Override
   public String put(StorageMetadata metadata, File localFile) {
-    throw new UnsupportedOperationException("Method not implemented");
+    hdfsAuth.authorize();
```

Comment:
I suppose not. Else there is no need for authorization.
`authorize()` throws an exception in case of an error. In that case the rest of the code path is not executed.